langchain4j/src/test/java/dev/langchain4j/service/AiServicesWithJsonSchemaIT.java

```diff
@@ -90,6 +93,245 @@ void should_extract_pojo_with_primitives() {
+                            !person.married
+            );
+
+            verify(model).chat(ChatRequest.builder()
+                    .messages(singletonList(userMessage(text)))
+                    .responseFormat(ResponseFormat.builder()
+                            .type(JSON)
+                            .jsonSchema(JsonSchema.builder()
+                                    .name("Set_of_Person")
+                                    .rootElement(JsonObjectSchema.builder()
+                                            .addProperty("array", JsonArraySchema.builder()
+                                                    .items(JsonObjectSchema.builder()
+                                                            .addStringProperty("name")
+                                                            .addIntegerProperty("age")
+                                                            .addNumberProperty("height")
+                                                            .addBooleanProperty("married")
+                                                            .required("name", "age", "height", "married")
+                                                            .build())
+                                                    .build())
+                                            .required("array")
+                                            .build())
+                                    .build())
+                            .build())
+                    .build());
+            verify(model).supportedCapabilities();
+        }
+    }
+
+    interface MaritalStatusExtractor {
+
+        enum MaritalStatus {
+
+            SINGLE, MARRIED
+        }
+
+        List<MaritalStatus> extractMaritalStatusFrom(String text);
+    }
+
+    @Test
+    void should_extract_list_of_enums() {
+
+        for (ChatLanguageModel model : models()) {
+
+            // given
+            model = spy(model);
+
+            MaritalStatusExtractor personExtractor = AiServices.create(MaritalStatusExtractor.class, model);
+
+            String text = "Klaus is 37 years old, 1.78m height and single. " +
+                    "Franny is 35 years old, 1.65m height and married." +
+                    "Staniel is 33 years old, 1.70m height and married.";
+
+            // when
+            List<MaritalStatusExtractor.MaritalStatus> statuses = personExtractor.extractMaritalStatusFrom(text);
+
+            // then
+            assertThat(statuses).hasSize(3);
+
+            // Use assertThat with filters to ensure each marital status is in the list
+            ArrayList<MaritalStatusExtractor.MaritalStatus> maritalStatuses = new ArrayList<>();
+            maritalStatuses.add(MaritalStatusExtractor.MaritalStatus.SINGLE);
+            maritalStatuses.add(MaritalStatusExtractor.MaritalStatus.MARRIED);
+            maritalStatuses.add(MaritalStatusExtractor.MaritalStatus.MARRIED);
+            Assertions.assertTrue(statuses.containsAll(maritalStatuses));
```

Comment:
this assertion can be improved:
`assertThat(statuses).containsExactly(SINGLE, MARRIED, MARRIED);`

Reply:
done